src/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -169,72 +129,82 @@ public static void Construction_MultiDimensionalArray()
+        public static IEnumerable<object[]> BinarySearch_Generic_TestData()
         {
-            int idx = Array.BinarySearch(array, value, comparer);
-            Assert.True(verifier(idx));
+            int[] intArray = new int[] { 1, 3, 6, 6, 8, 10, 12, 16 };
+
+            yield return new object[] { intArray, 0, 8, 8, null, 4 };
+            yield return new object[] { intArray, 0, 8, 6, null, 3 };
+            yield return new object[] { intArray, 0, 8, 12, null, 6 };
 
-            idx = Array.BinarySearch(array, value, genericComparer);
-            Assert.True(verifier(idx));
+            yield return new object[] { intArray, 0, 8, 0, null, -1 };
+            yield return new object[] { intArray, 0, 8, 99, null, ~intArray.Length };
 
-            idx = Array.BinarySearch(array, value);
-            Assert.True(verifier(idx));
+            yield return new object[] { intArray, 1, 5, 16, null, -7 };
         }
 
-        public static IEnumerable<object[]> BinarySearch_Range_TestData()
+        [Theory]
+        [MemberData(nameof(BinarySearch_Generic_TestData))]
+        [MemberData(nameof(BinarySearch_NonGeneric_TestData))]
+        public static void BinarySearch_NonGeneric(Array array, int index, int length, object value, IComparer comparer, int expected)
         {
-            var intArray = new int[] { 1, 3, 6, 6, 8, 10, 12, 16 };
-            IComparer intComparer = new IntegerComparer();
-            IComparer<int> genericIntComparer = new IntegerComparer();
-
-            var stringArray = new string[] { null, "aa", "bb", "bb", "cc", "dd", "ee" };
-            IComparer stringComparer = new StringComparer();
-            IComparer<string> genericStringComparer = new StringComparer();
-
-            yield return new object[] { intArray, 0, 8, 99, intComparer, genericIntComparer, new Func<int, bool>(i => i == ~(intArray.Length)) };
-            yield return new object[] { intArray, 0, 8, 6, intComparer, genericIntComparer, new Func<int, bool>(i => i == 2 || i == 3) };
-            yield return new object[] { intArray, 1, 5, 16, intComparer, genericIntComparer, new Func<int, bool>(i => i == -7) };
-            yield return new object[] { stringArray, 0, stringArray.Length, "bb", stringComparer, genericStringComparer, new Func<int, bool>(i => i == 2 || i == 3) };
-            yield return new object[] { stringArray, 3, 4, "bb", stringComparer, genericStringComparer, new Func<int, bool>(i => i == 3) };
-            yield return new object[] { stringArray, 4, 3, "bb", stringComparer, genericStringComparer, new Func<int, bool>(i => i == -5) };
-            yield return new object[] { stringArray, 4, 0, "bb", stringComparer, genericStringComparer, new Func<int, bool>(i => i == -5) };
-            yield return new object[] { stringArray, 0, 7, null, stringComparer, null, new Func<int, bool>(i => i == 0) };
+            bool isDefaultComparer = comparer == null || comparer == Comparer.Default;
```

Comment:
From the input data, it looks like comparer is always null?  What happened to the original test cases where it wasn't?